code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -272,6 +275,34 @@
 /obj/structure/chair/office/dark
 	icon_state = "officechair_dark"
 
+/obj/structure/chair/office/proc/GetArmrest()
+	return mutable_appearance('icons/obj/chairs.dmi', "[icon_state]_armrest")
+
+/obj/structure/chair/office/Initialize(mapload)
+	armrest = GetArmrest()
+	armrest.layer = ABOVE_MOB_LAYER
+	return ..()
+
+/obj/structure/chair/office/Destroy()
+	QDEL_NULL(armrest)
+	return ..()
+
+/obj/structure/chair/office/post_buckle_mob(mob/living/M)
+	. = ..()
+	update_armrest()
+
+/obj/structure/chair/office/post_unbuckle_mob(mob/living/M)
+	. = ..()
+	update_armrest()
+
+/obj/structure/chair/office/proc/update_armrest()
+	if(has_buckled_mobs())
+		add_overlay(armrest)
+	else
+		cut_overlay(armrest)
```

Comment:
can you rework this to use the `update_appearance()` system? I believe we have a flag for updating overlays, and that's generally the preferred way to do this.

Reply:
I'm not exactly a great coder, so I'm unsure if I could to be honest. Crunch had requested this PR.

Reply:
This is fine, honestly. I just like to encourage it, but I don't think it needs to happen, especially if it's how things already were.